### PR TITLE
feat: `ComboBox.IsEditable`

### LINKED ISSUE #9063: `ComboBox.IsEditable` is not implemented
### Current behavior

Currently WPF Combobox isEditable="true" does not allow for Combobox editing. This functionality works in UWP. Sample project is reused. Combobox is on first page. I have only tested WPF and UWP.

[ReactiveUIUnoSample.zip](https://github.com/unoplatform/uno/files/8919477/ReactiveUIUnoSample.zip)
 

### Expected behavior

Combobox to allow for editing when  isEditable="true".

### How to reproduce it (as minimally and precisely as possible)

-Create project
- Add Combobox w/ elements
- enable isEditable 
- Try to edit combobox item

### Workaround

n/a

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

WPF:
PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0"
PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0"
PackageReference Include="ReactiveUI.Fody" Version="17.1.17"
PackageReference Include="ReactiveUI.Uno" Version="17.*"
PackageReference Include="Uno.UI.Skia.Wpf" Version="4.4.0-dev.239"
PackageReference Include="Uno.UI.RemoteControl" Version="4.4.0-dev.241" Condition="'$(Configuration)'=='Debug'"

WPF.Host:
PackageReference Include="Uno.UI.Skia.Wpf" Version="4.4.0-dev.239"
PackageReference Include="Uno.UI.RemoteControl" Version="4.4.0-dev.241" Condition="'$(Configuration)'=='Debug'"

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.0.4

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #9063

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.cs [hunk 1]
```diff
@@ -1020,5 +1020,69 @@ public async Task When_VerticalContentAlignment_Is_Changed()
...
+				Width = 0,
+				Height = 0
+			};
+			textBox.Loaded += (s, e) => loaded = true;
+
+			WindowHelper.WindowContent = textBox;
+			await WindowHelper.WaitFor(() => loaded);
+
+			Assert.AreEqual(textBox.ActualWidth, 0);
+			Assert.AreEqual(textBox.ActualHeight, 0);
```
Comment (carldebilly) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.cs:1060]: It's weird, but the syntax for `Assert.AreEqual()` is `expected`, `actual`. It's one of the reason why I prefer FluentAssertions over the stock's Assert.AreEqual().

```suggestion
			Assert.AreEqual(0, textBox.ActualWidth);
			Assert.AreEqual(0, textBox.ActualHeight);
```

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs [hunk 2]
```diff
@@ -560,7 +560,15 @@ public bool Equals(LayoutBuilder other)
 
 			public void Build()
 			{
-				MeasuredSize = UpdateLayout(AvailableSize, _exactWidth);
+				if (AvailableSize.Width < 0 || AvailableSize.Height < 0)
+				{
+					// The layout is invalid, we can't build it.
+					MeasuredSize = new Size(0, 0);
+				}
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:567]: That's somewhat curious.

**Review by Youssef1313 — APPROVED** [review]
Quite a large PR, but assuming it's mostly porting from WinUI, LGTM

File: src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs:567]: Yes, most likely due to the fact that the ComboBox is keeping the text displayed but zero sized in some cases. So far it didn't cause any problems, but will not it to check

**Pull request merged**